### PR TITLE
NSF-501 - Automatically download CRL files during certificate verification

### PR DESCRIPTION
**Pull request by ansonj**
[NSF-501](https://themirror.atlassian.net/browse/NSF-501)

This PR tweaks OpenSSL so it works better for our purposes in a couple of ways:

- Most importantly: Copies CRL download code from `apps.c` (the CLI implementation of OpenSSL) into the C library code that we use, so that we can configure our `X509_STORE *` to automatically download CRL files when needed.
- Increases the internal HTTP client's max response length from 100 kb to 10 MB. One of our CRLs is already over 900 kb, so this change is necessary to allow OpenSSL to download our CRL file(s).
- Adds some logging to critical areas. When OpenSSL is compiled by [OpenSSL-SPM-LLS-fork](https://github.com/mirrorfitness/OpenSSL-SPM-LLS-fork), the logs also report what version is running, which helps make sure we're running the expected OpenSSL code.

[NSF-501]: https://themirror.atlassian.net/browse/NSF-501?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README
Comment (ansonj) [inline README:13]: This link is dead for now, until we finish merging this into MirrorLink. For now:

https://github.com/mirrorfitness/mirrorlink-ios/blob/NSF-501/README_SSL.md

File: crypto/ocsp/ocsp_ht.c
Comment (ansonj) [inline crypto/ocsp/ocsp_ht.c:34]: The max response length change as mentioned in the description.

File: crypto/x509/build.info
Comment (ansonj) [inline crypto/x509/build.info:11]: This adds a file to the build process, which helps us get things compiling without having to copy even more code in. I experimented with including all the source we needed by adding to this include list (instead of copying the functions over manually), but (1) that wasn't working, and it was a very tedious process of trial-and-error to see if everything was included as needed, and (2) it led to a lot of other files being compiled which aren't necessary. I was afraid of introducing unexpected changes due to adding a `#define` or something; I figured it was safer to manually copy over the most important functions, even though this is more tedious.

File: crypto/x509/x509_vfy.c
Comment (ansonj) [inline crypto/x509/x509_vfy.c:2554]: See: https://github.com/mirrorfitness/openssl-LLS-fork/blob/main-lululemon-studio/apps/apps.c

You can find the original function names by dropping the `lulu_studio_copies__` prefix.

File: crypto/x509/x509_vfy.c
Comment (ansonj) [inline crypto/x509/x509_vfy.c:2801]: On that note, I am adding a `!Thread.isMainThread` check to our client code to make sure this never runs on the main thread.

File: crypto/x509/x509_vfy.c
Comment (ansonj) [inline crypto/x509/x509_vfy.c:2831]: This function is exposed in the `x509_vfy.h` header file. We call this from our client code to activate the auto-download behavior.

File: crypto/x509/x509_vfy.c
Comment (vision360deg) [inline crypto/x509/x509_vfy.c:2560]: that makes sense; I agree it's safer to do a manual copy indeed

**Review by yuesaka-m — APPROVED** [review]
I'll defer to Federico if this requires some testing, but looks good to me -- thanks for the detailed README and the PR comments!

**Review by vision360deg — APPROVED** [review]
I was able to test it and see it working. Great work on this PR!

**Pull request merged**
